### PR TITLE
Store nav items in database

### LINKED ISSUE #62: As a maintainer, I want the links for navigation to be in a json/yaml, so that I can easily reuse them in different components without having to hardcode the data
### 📝 Description

I want to have a file containing all links and texts so that it can be reused in multiple components.
One example would be something like:
```yaml
Navigation:
  Sektionen:
    Link:
      URL: "/groups"
      Text: "Kommittéer, föreningar och andra instanser"
    Link:
      URL: "https://dokument.chalmers.it"
      Text: "Dokument"
  Link:
    URL: "/contact"
    Text: "Kontakt"

```

### ✅ Acceptance Criteria

-[] All navmenus use data from a file instead of hardcoded values.
-[] We have a file containing mappings of links, and dropdowns containing links.

### 💡 Additional context

_No response_

### PR DESCRIPTION
**Pull request by GAsplund**
# Key features
- Closes #62 
- Store all nav items in the database
- Settings UI for configuring nav items (admin only)

# Known issues
- There's really nothing stopping a nav category from having child items in the database, even if the category (aka "parent") has a URL (which is considered an illegal state, but child items will just be ignored should that occur)
- There is no "dynamic" CSS sizing on the navbar width, meaning that the content could look cramped before switching to hamburger menu
- The navbar will be blank when it's set up for the first time (meaning that there is no seeding script or similar to apply the previous design)

### CODE REVIEW

**Comment by GAsplund** [comment]
Okay, I'm not crazy. It seems like editing the navbar to be dynamic messes a lot with static routes. I think it propagates out to every other page but mainly is because (I think) that the root error page needs to be 100% static. I will look into this later when I have the time.

File: src/app/[locale]/settings/navbar/page.tsx
Comment (Oscariremma) [inline src/app/[locale]/settings/navbar/page.tsx]:
```suggestion
      <title>Control Panel - Navigation</title>
```
Use English to be consistent with the rest of the page.

Comment (Oscariremma) [reply src/app/[locale]/settings/navbar/page.tsx]: This does however seam to be a problem for all admin pages

File: src/app/[locale]/settings/navbar/page.tsx
Comment (Oscariremma) [inline src/app/[locale]/settings/navbar/page.tsx:1]: This works but is quite unintuitive as we can not see any changes until reloading the page. Is it possible to reload this page somehow when changing something? With my limited understanding I am guessing it might be difficult to pass that update out of the components?
It does not matter that much as the admin pages require a UI pass later any way but still...

**Review by Oscariremma — CHANGES_REQUESTED** [review]
The user facing functionality sems to work great but see comments on the admin page. 
Not necessarily required to be fixed before merging.

**Comment by GAsplund** [comment]
I've implemented the changes you requested - please retest

**Comment by Oscariremma** [comment]
The only thing I can think of now is that if you try to delete a category with elements it fails as a cascading delete is not performed. But this might be desirable to prevent accidental deletion?

Otherwise great!

**Comment by GAsplund** [comment]
> The only thing I can think of now is that if you try to delete a category with elements it fails as a cascading delete is not performed. But this might be desirable to prevent accidental deletion?
> 
> Otherwise great!

Good point, I had not thought of that. I think we discussed cases like this earlier at some point, and concluded that it might be best if we asked for confirmation stating what would happen, so I'll probably enable cascading deletions and do that

**Comment by simon-wg** [comment]
I'm unable to test-run this due to being away from my computer over the weekend. Just wanted to know if you've checked the mobile nav, everything should work fine from what I can see but you never know.

**Comment by GAsplund** [comment]
> I'm unable to test-run this due to being away from my computer over the weekend. Just wanted to know if you've checked the mobile nav, everything should work fine from what I can see but you never know.

Thanks for the reminder. I've checked, and it seems to work like before.

**Review by Oscariremma — APPROVED** [review]
LGTM

We need to make sure we are ready to recreate the existing nav elements when deploying this.

**Comment by GAsplund** [comment]
I agree. Let's hold off before pulling this on the server until we're ready to copy over

**Pull request merged**
